ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/metrics_correlation/MetricsCorrelation.java

```diff
@@ -0,0 +1,272 @@
+import org.opensearch.client.Client;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.ml.common.CommonValue;
+import org.opensearch.ml.common.FunctionName;
+import org.opensearch.ml.common.MLModel;
+import org.opensearch.ml.common.MLTask;
+import org.opensearch.ml.common.exception.MLException;
+import org.opensearch.ml.common.input.Input;
+import org.opensearch.ml.common.input.execute.metricscorrelation.MetricsCorrelationInput;
+import org.opensearch.ml.common.model.MLModelConfig;
+import org.opensearch.ml.common.model.MLModelFormat;
+import org.opensearch.ml.common.model.MLModelState;
+import org.opensearch.ml.common.model.MetricsCorrelationModelConfig;
+import org.opensearch.ml.common.output.execute.metrics_correlation.MCorrModelTensors;
+import org.opensearch.ml.common.output.execute.metrics_correlation.MetricsCorrelationOutput;
+import org.opensearch.ml.common.output.model.ModelResultFilter;
+import org.opensearch.ml.common.transport.deploy.MLDeployModelAction;
+import org.opensearch.ml.common.transport.deploy.MLDeployModelRequest;
+import org.opensearch.ml.common.transport.deploy.MLDeployModelResponse;
+import org.opensearch.ml.common.transport.model.MLModelGetAction;
+import org.opensearch.ml.common.transport.model.MLModelGetRequest;
+import org.opensearch.ml.common.transport.model.MLModelGetResponse;
+import org.opensearch.ml.common.transport.model.MLModelSearchAction;
+import org.opensearch.ml.common.transport.register.MLRegisterModelAction;
+import org.opensearch.ml.common.transport.register.MLRegisterModelInput;
+import org.opensearch.ml.common.transport.register.MLRegisterModelRequest;
+import org.opensearch.ml.common.transport.register.MLRegisterModelResponse;
+import org.opensearch.ml.common.transport.task.MLTaskGetAction;
+import org.opensearch.ml.common.transport.task.MLTaskGetRequest;
+import org.opensearch.ml.common.transport.task.MLTaskGetResponse;
+import org.opensearch.ml.engine.algorithms.DLModelExecute;
+import org.opensearch.ml.engine.annotation.Function;
+import org.opensearch.search.SearchHit;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BooleanSupplier;
+
+import static org.opensearch.index.query.QueryBuilders.termQuery;
+
+@Log4j2
+@NoArgsConstructor
+@Function(FunctionName.METRICS_CORRELATION)
+public class MetricsCorrelation extends DLModelExecute {
+
+    private static final int AWAIT_BUSY_THRESHOLD = 1000;
+    private Client client;
+    private Settings settings;
+
+    private static final String MODEL_NAME = "METRICS_CORRELATION";
+    public static final String MCORR_ML_VERSION = "1.0.0";
+
+    public static final String MCORR_MODEL_URL =
```

Comment:
Will we publish the model to model repo? Use github seems not good for production usage.

Reply:
Yeah we will publish the model to model repo. We are adding few more tests and changes in the model development repo. After that I will initiate the process to upload to model repo. Until then I'm using this URL as a placeholder for the development.